src/signed.rs

```diff
@@ -398,3 +403,90 @@ impl<'a, T: Signable, MK: MultiKeychain> PartiallyMultisigned<'a, T, MK> {
         }
     }
 }
+
+#[derive(Debug, Clone, Encode, Decode)]
+pub struct SignatureSet<S> {
+    pub signatures: NodeMap<Option<S>>,
+}
+
+impl<S: Signature> SignatureSet<S> {
+    pub(crate) fn new(len: NodeCount) -> Self {
+        SignatureSet {
+            signatures: NodeMap::new_with_len(len),
+        }
+    }
+}
+
+impl<S: Signature> PartialMultisignature for SignatureSet<S> {
+    type Signature = S;
+
+    fn add_signature(mut self, signature: &Self::Signature, index: NodeIndex) -> Self {
+        self.signatures[index] = Some(signature.clone());
+        self
+    }
+}
+
+/// Keybox wrapper which implements MultiKeychain such that a partial multisignature is a list of
+/// signatures and a partial multisignature is considered complete if it contains more than 2N/3 signatures.
+///
+/// Note: this way of multisigning is very inefficient, and should be used only for testing.
+#[derive(Debug, Clone)]
+pub struct DefaultMultiKeychain<KB: KeyBox> {
+    key_box: KB,
+}
+
+impl<KB: KeyBox> DefaultMultiKeychain<KB> {
+    pub fn new(key_box: KB) -> Self {
+        DefaultMultiKeychain { key_box }
+    }
+}
+
+impl<KB: KeyBox> Index for DefaultMultiKeychain<KB> {
+    fn index(&self) -> NodeIndex {
+        self.key_box.index()
+    }
+}
+
+#[async_trait::async_trait]
+impl<KB: KeyBox> KeyBox for DefaultMultiKeychain<KB> {
+    type Signature = KB::Signature;
+
+    async fn sign(&self, msg: &[u8]) -> Self::Signature {
+        self.key_box.sign(msg).await
+    }
+
+    fn node_count(&self) -> NodeCount {
+        self.key_box.node_count()
+    }
+
+    fn verify(&self, msg: &[u8], sgn: &Self::Signature, index: NodeIndex) -> bool {
+        self.key_box.verify(msg, sgn, index)
+    }
+}
+
+impl<KB: KeyBox> MultiKeychain for DefaultMultiKeychain<KB> {
+    type PartialMultisignature = SignatureSet<KB::Signature>;
+
+    fn from_signature(
+        &self,
+        signature: &Self::Signature,
+        index: NodeIndex,
+    ) -> Self::PartialMultisignature {
+        SignatureSet::add_signature(SignatureSet::new(self.node_count()), signature, index)
+    }
+
+    fn is_complete(&self, msg: &[u8], partial: &Self::PartialMultisignature) -> bool {
+        let signature_count = partial
+            .signatures
+            .iter()
+            .filter(|sgn| sgn.is_some())
+            .count();
+        if signature_count <= 2 * self.node_count().0 / 3 {
```

Comment:
It would be good to have this as a threshold parameter or calculated by a global function like `minimal_quorum` or sth like that.

Reply:
Added a `quorum` method